manager-bundle/src/DependencyInjection/Compiler/ParametersYamlPass.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class ParametersYamlPass implements CompilerPassInterface
+{
+    /**
+     * @var ContainerBuilder
+     */
+    private $container;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container): void
+    {
+        $this->container = $container;
+
+        // The "mail" transport has been removed in SwiftMailer 6, so use "sendmail" instead
+        if ('mail' === $this->container->getParameter('mailer_transport')) {
+            $this->container->setParameter('mailer_transport', 'sendmail');
+        }
+
+        if (!isset($_SERVER['APP_SECRET'])) {
+            $this->container->setParameter('env(APP_SECRET)', $this->getAppSecret());
+        }
+
+        if (!isset($_SERVER['DATABASE_URL'])) {
+            $this->container->setParameter('env(DATABASE_URL)', $this->getDatabaseUrl());
+        }
+
+        if (!isset($_SERVER['MAILER_URL'])) {
+            $this->container->setParameter('env(MAILER_URL)', $this->getMailerUrl());
+        }
+    }
+
+    private function getAppSecret(): string
+    {
+        return $this->container->getParameter('secret');
```

Comment:
What if the parameter is not defined at all?

Reply:
It is not possible to have a `parameters.yml` file in Contao <4.9 without a secret, so if there is no secret at all, the app will crash in all Contao 4 versions.

Reply:
I thought the secret is generated in the install tool, and the parameter has a default value so the app does **not** crash when the install tool is run?

Reply:
I don't understand what you are up to? The parameter is always defined, which should answer your initial question.